formulaire/src/main/resources/public/ts/controllers/respond-question.ts

```diff
@@ -229,6 +249,12 @@ export const respondQuestionController = ng.controller('RespondQuestionControlle
                     await responseService.create(new Response(child.id, null, null, vm.distribution.id));
                 }
             }
+            // In case of question type ranking, we need to add a choice index to each Response
+            if (question.isRanking()) {
+                for (let resp of responses.all) {
+                    await responseService.create(new Response(question.id, resp.choice_id, resp.answer, vm.distribution.id,  resp.choice_index));
```

Comment:
Dans ta boucle tu fais un await sans vraiment faire un traitement de retour suite aux réponses => est-ce qu'au final c'est pertinent de faire un await ?

A voir si on peut enlever l'opération `await`

```suggestion
                responseService.create(new Response(question.id, resp.choice_id, resp.answer, vm.distribution.id,  resp.choice_index));
```


Si malgré tout tu souhaites utiliser l'opération, ça serait intéréssant d'utiliser `Promise.all` pour pouvoir agréger toutes tes promises d'un coup (cf [doc](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)) :  

```
const promises: Promise<any>[] = [];
for (let resp of responses.all) {
   promises.push(responseService.create(new Response(question.id, resp.choice_id, resp.answer, vm.distribution.id,  resp.choice_index)));
}
let result: any[] = await Promise.all(promises);      
```